gossip/src/crds.rs

```diff
@@ -303,8 +311,21 @@ impl Crds {
                     Err(CrdsError::InsertFailed)
                 } else if matches!(route, GossipRoute::PushMessage(_)) {
                     let entry = entry.get_mut();
-                    entry.num_push_dups = entry.num_push_dups.saturating_add(1);
-                    Err(CrdsError::DuplicatePush(entry.num_push_dups))
+                    // If num_push_recv == 0, we know this data was first received
+                    // via PullResponse. This is a redundant pull, meaning we received
+                    // a PushMessage after we had already received the same message
+                    // first via PullResponse
```

Comment:
I _think_ this is not true if the value was upserted from `GossipRoute::LocalMessage` or `GossipRoute::PullRequest` as in:
https://github.com/anza-xyz/agave/blob/0f1ca20d3/gossip/src/crds_gossip_pull.rs#L296
So this will over count redundant pull messages.